src/analysis/inner_product_space/adjoint.lean

```diff
@@ -186,6 +186,29 @@ lemma is_adjoint_pair (A : E' →L[ℝ] F') :
 
 end real
 
+section unitary
+
+variables {U : (E →L[𝕜] E)}
+
+/-- Builds a linear isometric equivalence from `U ∈ unitary (E →L[𝕜] E)`. -/
+def linear_isometry_equiv_of_unitary (hU : U ∈ unitary (E →L[𝕜] E)) : (E ≃ₗᵢ[𝕜] E) :=
+{ to_linear_equiv :=
+  { to_fun    := U,
+    map_add'  := map_add U,
+    map_smul' := map_smul U,
+    inv_fun   := (star U : (E →L[𝕜] E)),
+    left_inv  := λ x, by rw [←mul_apply, unitary.star_mul_self_of_mem hU, one_apply],
+    right_inv := λ x, by rw [←mul_apply, unitary.mul_star_self_of_mem hU, one_apply] },
+  norm_map' := λ x, by
+  { rw unitary.mem_iff at hU,
+    rw [linear_equiv.coe_mk, ←sq_eq_sq (norm_nonneg (U x)) (norm_nonneg x), norm_sq_eq_inner,
+      ←adjoint_inner_left, ←mul_apply, ←star_eq_adjoint, hU.1, one_apply, inner_self_eq_norm_sq] } }
```

Comment:
A little shorter:
```suggestion
{ to_linear_equiv :=
    linear_map.general_linear_group.to_linear_equiv $
      units.map (continuous_linear_map.to_linear_map_hom : (E →L[𝕜] E) →* _)
        (unitary.to_units ⟨U, hU⟩),
  norm_map' := λ x,
    show ∥U x∥ = ∥x∥,
    by rw [←sq_eq_sq (norm_nonneg (U x)) (norm_nonneg x), norm_sq_eq_inner,
      ←adjoint_inner_left, ←mul_apply, ←star_eq_adjoint, unitary.star_mul_self_of_mem hU, one_apply,
      inner_self_eq_norm_sq] }
```
which also nicely demonstrates all the pieces this factors into

Reply:
Can you clarify what you mean by "somewhere sensible"?  I still feel cautious about dumping things into mathlib, and so I'd rather not dump it in the wrong place.

Reply:
I've inserted it here, but I'd appreciate some guidance on how best to find a home for your `simps` def

Reply:
It probably belongs shortly after wherever the monoid structure it refers to is defined (I don't know where that is, but you can get vscode to tell you by hovering over the goal view for this lemma)